graphistry/text_utils.py

```diff
@@ -270,6 +271,7 @@ def search_graph(
 
     def save_search_instance(self, savepath):
         from joblib import dump  # type: ignore   # need to make this onnx or similar
+        self.build_index()
```

Comment:
maybe we make `self.search_index` a lazy auto-memoizing getter? i'm not sure when it invalidates, but the peppering of `build_index()` sounds like we're doing that anyway, so maybe centralize